src/components/EditableHeading/EditableHeading.module.scss

```diff
@@ -0,0 +1,36 @@
+@import "../../styles/typography";
+@import "~monday-ui-style/dist/mixins";
+
+@include create-title-classes;
+
+.editableHeading {
```

Comment:
I took this example from the modal with the editable header. 
It should be aligned the same as the description text, and when hovered, the border should be applied outside this line. you can talk with Meirav if it is not clear

<img width="999" alt="image" src="https://github.com/mondaycom/monday-ui-react-core/assets/6093192/834e3047-c0c5-493b-bb8f-cd9c730e6018">

Reply:
I'll make sure with Meirav, the old component behaves the same, if we shift it to the left it means the the input will overflow

Reply:
The input should probably contain some padding so the text inside would be aligned, but the input frame will overflow

Reply:
Yes but it might be a big deal to shift it in `overflow: hidden` containers like this one, we'll have to use a portal for this
![Screenshot 2023-11-01 at 12 45 58](https://github.com/mondaycom/monday-ui-react-core/assets/18269880/f5d80d07-a13e-4d8b-8930-631fd4fcd156)

Reply:
Or we can solve this with some padding games. Give the content some padding and reduce it from the container

Reply:
Wouldn't that be too hacky? It means I'll have to adjust the `Modal*` components' width just for this case. And it would have to be adjusted as well in other overflowed containers

Reply:
Yes, it is an issue that will appear in any container that is overflowing, and that the content sticks to the side.
But I don't think it's that hacky... it's just how you split the padding. it might get complex if all the rendered elements (header, description, content) aren't wrapped with any wrapper from the component.

But let's give it a shot and see if it goes bad or not

Reply:
I added an opt-out prop (`alignWithText` - if you have a better name LMK) that adds negative margin to the component so it will be aligned with the text. In case of hidden overflowing containers users can disable it. Specifically the `ModalHeader` has a different issue when providing a custom title but the might be a breaking change, I suggest discussing it further